### ISSUE TITLE
unused-import (W0611) false positive when imported type used only as :rtype: in docstrings

### ISSUE THREAD

**Issue by DeepSpace2** [PyCQA/pylint#2947]
### Steps to reproduce
1. Import a type
2. Use it in a docstring in `:rtype:`

### Current behavior

We get W0611 unused-import

### Expected behavior

We should not get unused-import

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 14:57:15) [MSC v.1915 64 bit (AMD64)]
```

### Demo

```python
from car import Car


def get_car(list_of_cars):
    """
    :rtype: Car
    :return:
    """
    return list_of_cars[0]
````
```
$ pylint main.py
...
W0611: Unused Car imported from car (unused-import)
```

**Comment by PCManticore** [comment]
pylint doesn't care about imports being used in docstrings or not. If you want to exempt these imports from `unused-import`, you should use `typing.TYPE_CHECKING` to signify that they are imported just for type hinting, albeit type hinting in a docstring:
```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from collections import defaultdict

def get_car(list_of_cars):
    """
    :rtype: defaultdict
    :return:
    """
    return list_of_cars[0]
```
